esp_wrover_kit/include/bsp/display.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+/**
+ * @file
+ * @brief BSP LCD
+ *
+ * This file offers API for basic LCD control.
+ * It is useful for users who want to use the LCD without a Graphical Library or without the default LVGL library.
+ *
+ * For standard LCD initialization, you can call all-in-one function bsp_display_start().
+ */
+
+#pragma once
+#include "esp_lcd_types.h"
+
+/**
+ * @brief BSP display configuration structure
+ */
+typedef union {
+    struct {
+        int num;               /*!< Index of SPI peripheral to use */
+        int clk_hz;            /*!< SPI clock in Hz */
+        int max_transfer_size; /*!< SPI max transfer size in bytes */
+    } spi;
+} bsp_display_config_t;
+
+/**
+ * @brief Create new display panel
+ *
+ * For maximum flexibility, this function does not perform the usual reset-init-turn-on sequence of the display.
+ * The display's backlight is not turned on either. You can use bsp_display_brightness_init() and bsp_display_backlight_on(),
+ * or implement your own backlight control.
+ *
+ * If you want to free resources allocated by this function, you can use esp_lcd API, ie.:
+ *
+ * \code{.c}
+ * esp_lcd_panel_del(panel);
+ * esp_lcd_panel_io_del(io);
+ * spi_bus_free(spi_num_from_configuration);
+ * \endcode
+ *
+ * @param[in]  config    Configuration of display HAL
+ * @param[out] ret_panel esp_lcd panel handle
+ * @param[out] ret_io    esp_lcd IO handle
+ * @return
+ *      - ESP_OK                On success
+ *      - Else                  esp_lcd failure
+ */
+esp_err_t bsp_display_new(const bsp_display_config_t *config, esp_lcd_panel_handle_t *ret_panel, esp_lcd_panel_io_handle_t *ret_io);
```

Comment:
I think, that this function should be independent. Without `config`. 
If you can change the board with another LCD, then it will be very simple - there should be same API without any other settings.

Reply:
Yes, that might be better. 

However, we should implement a way of changing these HAL parameters (SPI num, clk and max_transfer_size). Currently they are hardcoded in the header, which cannot be changed if you are using idf-component-manager.

Would it be OK to change it via menuconfig?

Reply:
Yes, we should put these params into menuconfig.